rdrf/rdrf/export/model_exporters.py

```diff
@@ -0,0 +1,72 @@
+import os
+from rdrf import models
+from registry.patients import models as patientmodels
+from .catalogue import ModelExporterCatalogue
+from .exporters import ModelExporter
+
+
+# Catalogue with generic exporters
+catalogue = ModelExporterCatalogue()
+
+# Registry specific exporters. Querysets filtered by registry code.
+registry_catalogue = ModelExporterCatalogue()
+
+class RegistryExporter(ModelExporter):
+    @property
+    def queryset(self):
+        return self.model.objects.filter(code=self.exporter_context['registry_code'])
+
+
+class RegistryFormExporter(ModelExporter):
+    @property
+    def queryset(self):
+        return self.model.objects.filter(registry__code=self.exporter_context['registry_code'])
+
+
+class AdjudicationDefinitionExporter(ModelExporter):
+    @property
+    def queryset(self):
+        return self.model.objects.filter(registry__code=self.exporter_context['registry_code'])
+
+
+class SectionExporter(ModelExporter):
+    @staticmethod
+    def get_sections(registry_code):
+        registry = models.Registry.objects.get(code=registry_code)
+        for form in models.RegistryForm.objects.filter(registry=registry):
+            for section_code in form.get_sections():
+                yield models.Section.objects.get(code=section_code)
+        if registry.patient_data_section is not None:
+            yield registry.patient_data_section
+        for section_code in registry.generic_sections:
+            yield models.Section.objects.get(code=section_code)
+        for adj_def in models.AdjudicationDefinition.objects.filter(registry=registry):
+            yield models.Section.objects.get(code=adj_def.result_fields)
+            yield models.Section.objects.get(code=adj_def.decision_field)
+
+    @property
+    def queryset(self):
+        return self.get_sections(self.exporter_context['registry_code'])
+
+
+class CommonDataElementExporter(ModelExporter):
+    @property
+    def queryset(self):
+        sections = SectionExporter.get_sections(self.exporter_context['registry_code'])
+        for cde in reduce(lambda res, sec: res.union(sec.cde_models), sections, set()):
+            yield cde
+
+
+class PatientExporter(ModelExporter):
+    @property
+    def queryset(self):
+        return self.model.objects.filter(rdrf_registry__code=self.exporter_context['registry_code'])
+
+
+registry_catalogue.register(models.Registry, RegistryExporter)
+registry_catalogue.register(models.RegistryForm, RegistryFormExporter)
+registry_catalogue.register(models.AdjudicationDefinition, AdjudicationDefinitionExporter)
+registry_catalogue.register(models.Section, SectionExporter)
+registry_catalogue.register(models.CommonDataElement, CommonDataElementExporter)
+
+registry_catalogue.register(patientmodels.Patient, PatientExporter)
```

Comment:
Will need exporters for:
QuestionnaireResponse (used by DM1),
Query (reports)
CdePolicy
ConsentSection
ConsentQuestion
ConsentValue
DemographicFields
PatientRelative
EmailTemplate
EmailNotification
EmailNotificationHistory
RDRFContext